Automattic/wp-calypso
# Playwright: Implement video capture.

#### Changes proposed in this Pull Request

This patch implements video capture using Playwright 1.11.0.

- implements video capture using native Playwright API calls wrapped around hooks and additional processing. 
- uses a root-level `before` and `after` hook to start/terminate browser instances.

Implementation Details

- as of 1.11.0, Playwright is able to delete video files for a given Page object. It is also possible to provide a custom name when saving the video. However, my investigations point to there being a bug where `saveAs` writes two copies of the same file, one with original name and one with custom name. This requires additional processing in the case of failing tests, to manually remove the duplicate file.
- an array instantiated at the suite-level hook is used to track failures occurring in a given `mocha` run. Since we are using `magellan` in conjunction with `mocha`, each suite tagged with `@parallel` will execute in an isolated mocha environment. This means the array is guaranteed to contain only one item.

#### Testing instructions

Demo spec `wp-log-in-out-spec.js` is failing by default; ensure TeamCity is able to find and upload the recording to the Artifacts tab.

Partially implements #51082.
Parent: #51209
Child: #52843
